Cn 886 support docker image artifact

billiford opened the pull request:
This PR:
- Adds support for the `docker/image` artifact type

dmrogers7 commented on pkg/kubernetes/artifact.go:
```diff
@@ -0,0 +1,70 @@
+		return nil
+	}
+
+	if strings.EqualFold(u.GetKind(), "pod") {
+		containers, found, err = unstructured.NestedSlice(u.Object, "spec", "containers")
+	} else {
+		containers, found, err = unstructured.NestedSlice(u.Object, "spec", "template", "spec", "containers")
```
I failed to give full requirements in the story, but will need to get `initContainers` as well

See the [clouddriver code](https://github.com/spinnaker/clouddriver/blob/58ab154b0ec0d62772201b5b319af349498a4e3f/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/Replacer.java#L166)

billiford replied:
Arg! I missed that.

dmrogers7 commented on pkg/kubernetes/artifact_test.go:
```diff
@@ -0,0 +1,153 @@
+		})
+
+		When("the kind is deployment", func() {
+			BeforeEach(func() {
```
Can you write a `when a container image does not match artifact image` test?  Define the resource with two containers/images and validate that only the one image is replaced

```go
		resource = &unstructured.Unstructured{
			Object: map[string]interface{}{
				"kind":       "Deployment",
				"apiVersion": "apps/v1",
				"spec": map[string]interface{}{
					"template": map[string]interface{}{
						"spec": map[string]interface{}{
							"containers": []interface{}{
								map[string]interface{}{
									"name":  "test-container-name",
									"image": "gcr.io/test-project/test-container-image",
								},
								map[string]interface{}{
									"name":  "another-test-container-name",
									"image": "gcr.io/test-project/another-test-container-image",
								},
							},
						},
					},
				},
			},
		}
	})		
```

billiford replied:
Sure thing.

billiford commented:
- changed function name to `BindDockerImageArtifacts`
- added support for init containers
- added full test section for init containers
- added new tests for multiple containers/initContainers

dmrogers7 reviewed (approved):
LGTM!
